rts/lra/lra-client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java

```diff
@@ -909,12 +895,12 @@ private boolean isStatus(URL lraId, CompensatorStatus testStatus) {
      * @return the corresponding status or empty
      * @throws IllegalArgumentException if the status is not a valid enum constant
      */
-    private static Optional<CompensatorStatus> fromString(String status) {
+    private static LRAStatus fromString(String status) {
         if (status != null) {
-            return Optional.of(CompensatorStatus.valueOf(status));
+            return LRAStatus.valueOf(status);
         }
 
-        return Optional.empty();
+        return null;
```

Comment:
The status should never be null, instead it should either throw generic exception (including NotFound if that is what the response from the coordinator indicated).

Reply:
ok, agree. I think the method `fromString(...)` should throw the `IllegalArgumentException` as defined in its javadoc